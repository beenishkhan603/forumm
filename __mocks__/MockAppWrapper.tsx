import { AuthContext } from '@libs/useAuth'
import { MockedProvider, MockedResponse } from '@apollo/client/testing'
import { ProfileInfo } from '@type/Hooks/useAuth/Auth.type'
import ALL_MOCKS from './graphql/mocks'

export const defaultMockProfile: ProfileInfo = {
  company: '448 Studio',
  email: 'test@email.com',
  fullName: 'John Doe',
  university: '448 Studio',
  profileImageUrl: '/test.png',
  userId: '123',
  groups: [],
}

const MockAppWrapper = ({
  children,
  mocks,
  mockProfile = defaultMockProfile,
}: {
  mockProfile?: Partial<ProfileInfo>
  children: any
  mocks?: readonly MockedResponse<Record<string, any>>[] | undefined
}) => {
  const Layout = children?.type?.Layout ?? null

  const GQL_MOCKS: MockedResponse[] = ALL_MOCKS

  if (mocks) mocks.forEach((m) => GQL_MOCKS.push(m))
  return (
    <MockedProvider
      mocks={GQL_MOCKS}
      defaultOptions={{
        query: {
          errorPolicy: 'all',
          fetchPolicy: 'cache-first',
        },
        watchQuery: {
          errorPolicy: 'all',
          fetchPolicy: 'cache-and-network',
        },
        mutate: {
          errorPolicy: 'all',
        },
      }}
    >
      <AuthContext.Provider
        value={{
          profile: mockProfile
            ? { ...defaultMockProfile, ...mockProfile }
            : defaultMockProfile,
          setProfile: () => {},
        }}
      >
        <div id="modal-root" />
        {Layout ? <Layout>{children}</Layout> : children}
      </AuthContext.Provider>
    </MockedProvider>
  )
}

export default MockAppWrapper
