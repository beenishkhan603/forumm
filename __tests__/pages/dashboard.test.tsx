import {
  render,
  waitFor,
  screen,
  waitForElementToBeRemoved,
} from '@testing-library/react'
import OrganizerLaunchPage from 'pages/dashboard'
import MockAppWrapper, { defaultMockProfile } from '__mocks__/MockAppWrapper'

it('renders dashboard page unchanged when not a forumm admin', async () => {
  const { container } = render(
    <MockAppWrapper>
      <OrganizerLaunchPage />
    </MockAppWrapper>
  )

  await waitForElementToBeRemoved(() => screen.queryAllByTestId('loading'))

  waitFor(() => {
    expect(container).toMatchSnapshot()
  })
})

// This test is not required because the page is rendered differently for forumm admins
/* it('renders dashboard page unchanged when a forumm admin', async () => {
  const { container } = render(
    <MockAppWrapper
      mockProfile={{
        ...defaultMockProfile,
        groups: ['forumm-admin'],
      }}
    >
      <OrganizerLaunchPage />
    </MockAppWrapper>
  )
  await waitForElementToBeRemoved(() => screen.queryAllByTestId('loading'))

  waitFor(() => {
    expect(container).toMatchSnapshot()
  })
})*/
