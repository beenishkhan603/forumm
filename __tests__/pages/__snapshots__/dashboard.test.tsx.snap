// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders dashboard page unchanged when not a forumm admin 1`] = `
<div>
  <div
    id="modal-root"
  />
  <div
    class="overflow-hidden min-h-screen flex flex-col h-screen"
    style="color: rgb(24, 26,12);"
  >
    <header>
      <div
        class="sticky top-0 left-0 w-full z-[1000] flex px-4 md:px-20 lg:px-[156px] justify-between border-b border-forumm-menu-border  items-center h-[65px] sm:h-[75px]"
        style="color: rgb(24, 26,12); background: rgba(255, 255, 255, 0.6);"
      >
        <div
          class="flex items-center"
          style="color: rgb(24, 26,12);"
        >
          <button>
            <div
              class="flex items-center hover:animate-heartbeat"
              style="color: rgb(24, 26,12);"
            >
              <img
                alt="Logo"
                class="hidden md:flex h-14 w-14 rounded-md object-cover mr-2"
                data-nimg="1"
                decoding="async"
                height="100"
                loading="lazy"
                src="/images/ForummLogo.svg"
                srcset="/images/ForummLogo.svg 1x, /images/ForummLogo.svg 2x"
                style="color: transparent;"
                width="100"
              />
              <div
                class="flex flex-col"
                style="color: rgb(24, 26,12);"
              >
                <svg
                  class="w-20 h-8 -ml-1"
                />
                <span
                  class="text-xs text-left min-h-[20px]"
                >
                  448 Studio
                </span>
              </div>
            </div>
          </button>
          <div
            class="hidden md:block ml-6 font-medium text-xl"
            style="color: rgb(24, 26,12);"
          />
        </div>
        <div
          class="flex space-x-4 items-center cursor-pointer"
          style="color: rgb(24, 26,12);"
        >
          <div
            class="flex items-center space-x-2 md:space-x-6 mr-3 md:mr-6"
          >
            <span
              class="border-l-2 border-textColour h-8"
            />
          </div>
        </div>
      </div>
      <div
        class=" w-64 absolute px-6 right-0 mr-7 flex flex-col space-y-2 text-lg transition-all h-0 overflow-hidden -mt-[1px] rounded-xl border-t-0 rounded-tl-none rounded-tr-none shadow-md"
        style="z-index: 999999; background-color: rgb(251, 252, 254); border: 0px solid #e5e7eb;"
      >
        <div
          class="cursor-pointer flex items-center textColour text-base leading-none"
          style="color: rgb(24, 26,12);"
        >
          <svg
            class="inline w-5 h-5 mr-2"
            fill="#181A20"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 496 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M248 8C111 8 0 119 0 256s111 248 248 248 248-111 248-248S385 8 248 8zm0 96c48.6 0 88 39.4 88 88s-39.4 88-88 88-88-39.4-88-88 39.4-88 88-88zm0 344c-58.7 0-111.3-26.6-146.5-68.2 18.8-35.4 55.6-59.8 98.5-59.8 2.4 0 4.8.4 7.1 1.1 13 4.2 26.6 6.9 40.9 6.9 14.3 0 28-2.7 40.9-6.9 2.3-.7 4.7-1.1 7.1-1.1 42.9 0 79.7 24.4 98.5 59.8C359.3 421.4 306.7 448 248 448z"
            />
          </svg>
          <div
            style="color: rgb(24, 26,12);"
          >
            User Profile
          </div>
        </div>
        <div
          class="cursor-pointer flex items-center textColour text-base pt-1 leading-none"
          style="color: rgb(24, 26,12);"
        >
          <svg
            class="inline w-5 h-5 mr-2 text-181A20"
          />
          <div
            style="color: rgb(24, 26,12);"
          >
            Support
          </div>
        </div>
        <div
          class="cursor-pointer flex items-center textColour pt-1 text-base leading-none"
          style="color: rgb(24, 26,12);"
        >
          <svg
            class="inline w-5 h-5 mr-2"
            fill="#181A20"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 0h24v24H0V0z"
              fill="none"
            />
            <path
              d="M17 8l-1.41 1.41L17.17 11H9v2h8.17l-1.58 1.58L17 16l4-4-4-4zM5 5h7V3H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h7v-2H5V5z"
            />
          </svg>
          <div
            style="color: rgb(24, 26,12);"
          >
            Logout
          </div>
        </div>
      </div>
    </header>
    <div
      class="flex flex-col flex-1"
      style="color: rgb(24, 26,12);"
    >
      <div
        class="flex flex-1"
        style="color: rgb(24, 26,12);"
      >
        <div
          class="flex-1 overflow-y-scroll h-[calc(100vh)] relative flex items-center flex-col"
          style="color: rgb(24, 26,12);"
        >
          <div
            class="w-full h-2 bg-blue-100 overflow-hidden relative"
          >
            <div
              class="w-full h-full absolute transform translate-x-0 bg-gradient-to-r from-transparent via-blue-500 to-transparent"
              style="background-size: 200% 100%; animation: shimmer 1.5s infinite;"
            />
          </div>
        </div>
        <div
          class="w-0 border-l border-panel-gray transition-all flex flex-col overflow-x-hidden h-[calc(100vh_-_81px)]"
          style="color: rgb(24, 26,12);"
        >
          <div
            class="flex justify-evenly items-center border-b border-panel-gray text-center"
            style="color: rgb(24, 26,12);"
          >
            <div
              class="text-md font-medium py-4 flex-1 cursor-pointer select-none whitespace-nowrap"
              style="background-color: rgb(55, 99, 233); color: white;"
            >
              448 Studio Chat
            </div>
            <div
              class="text-white text-sm py-4 px-6 cursor-pointer select-none flex items-center space-x-2 justify-center"
              style="background-color: rgb(255, 255, 255);"
            >
              <svg
                class="w-5 h-6"
                fill="#181A20"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M0 0h24v24H0z"
                  fill="none"
                />
                <path
                  d="M20 2H4c-1.1 0-1.99.9-1.99 2L2 22l4-4h14c1.1 0 2-.9 2-2V4c0-1.1-.9-2-2-2zm-2 12H6v-2h12v2zm0-3H6V9h12v2zm0-3H6V6h12v2z"
                />
              </svg>
            </div>
            <div
              class="text-white text-sm py-4 px-6 cursor-pointer select-none flex items-center space-x-2 justify-center"
              style="background-color: rgb(255, 255, 255);"
            >
              <svg
                class="w-5 h-6"
                fill="#181A20"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M0 0h24v24H0z"
                  fill="none"
                />
                <path
                  d="M16 11c1.66 0 2.99-1.34 2.99-3S17.66 5 16 5c-1.66 0-3 1.34-3 3s1.34 3 3 3zm-8 0c1.66 0 2.99-1.34 2.99-3S9.66 5 8 5C6.34 5 5 6.34 5 8s1.34 3 3 3zm0 2c-2.33 0-7 1.17-7 3.5V19h14v-2.5c0-2.33-4.67-3.5-7-3.5zm8 0c-.29 0-.62.02-.97.05 1.16.84 1.97 1.97 1.97 3.45V19h6v-2.5c0-2.33-4.67-3.5-7-3.5z"
                />
              </svg>
            </div>
          </div>
          <div
            class="flex flex-col h-full overflow-y-auto"
            style="color: rgb(24, 26,12);"
          >
            <div
              class="relative w-full h-full flex flex-col"
              style="color: rgb(24, 26,12);"
            >
              <div
                class="flex-1 flex flex-col overflow-y-auto pt-4 pb-2"
                style="color: rgb(24, 26,12);"
              />
              <div
                class="border-t border-gray-300"
                style="color: rgb(24, 26,12);"
              />
              <div
                class="flex-end w-full"
                style="color: rgb(24, 26,12);"
              >
                <div
                  class="flex w-full p-3 space-x-1 items-center"
                  style="color: rgb(24, 26,12);"
                >
                  <label
                    class="flex flex-col flex-1 mt-0"
                  >
                    <span
                      class="absolute text-sm -translate-y-1/2 mb-2 ml-2 px-2 transition-all "
                      style="background-color: rgb(255, 255, 255);"
                    >
                      <span
                        class=""
                      />
                    </span>
                    <textarea
                      autocomplete="on"
                      class="transition-all bg-transparent   resize-none rounded px-4 py-4 text-sm outline-none"
                      placeholder="Type to write a message..."
                      rows="1"
                    />
                    <div
                      class="transition-all opacity-0 text-red-500 text-sm font-bold"
                    />
                  </label>
                  <span>
                    <svg
                      class="text-2xl cursor-pointer text-black"
                      fill="currentColor"
                      height="1em"
                      stroke="currentColor"
                      stroke-width="0"
                      viewBox="0 0 16 16"
                      width="1em"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M4 8a.5.5 0 0 1 .5-.5h5.793L8.146 5.354a.5.5 0 1 1 .708-.708l3 3a.5.5 0 0 1 0 .708l-3 3a.5.5 0 0 1-.708-.708L10.293 8.5H4.5A.5.5 0 0 1 4 8z"
                        fill-rule="evenodd"
                      />
                    </svg>
                  </span>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
